library/src/main/kotlin/com/connectrpc/impl/ProtocolClient.kt

```diff
@@ -211,6 +211,9 @@ class ProtocolClient(
                 }
             }
             channel.send(result)
+            if (isComplete) {
+                channel.close()
```

Comment:
This was a major issue - not closing the channel would lead to hangs in streaming clients trying to read results.